book/scheduling-and-threading.md

```diff
@@ -233,24 +177,26 @@ function setTimeout(callback, time_delta) {
     SET_TIMEOUT_REQUESTS[handle] = callback;
     call_python("setTimeout", handle, time_delta)
 }
+```
+
+The exported `setTimeout` function will create a timer, wait for the
+requested time period, and then call back into JavaScript to run the
+callback. The call back can call this `__runSetTimeout` function:
 
+``` {.javascript file=runtime}
 function __runSetTimeout(handle) {
     var callback = SET_TIMEOUT_REQUESTS[handle]
     delete SET_TIMEOUT_REQUESTS[handle];
     callback();
 }
 ```
 
-On the Python side, add a binding for `setTimeout` and an implementation that
-starts a `threading.Timer`. However, we have to be careful here, since in the
-code below, `run_callback` will run *on a different Python thread than the
-current one*. So we can't just call `evaljs` directly, or we'll end up with
-JavaScript running on two Python threads at the same time, which is not
-ok.[^js-thread]
-
-This is easy to fix by using the `TaskRunner` you already implemented: instead
-of running the script right away, schedule a task to do it later, when the
-other thread is free. Here's the code:
+The Python side, however, is quite a bit more complex, because
+`threading.Timer` executes its callback *on a new Python thread*. That
+thread can't just call `evaljs` directly: we'll end up with JavaScript
+running on two Python threads at the same time, which is not
+ok.[^js-thread] Instead, the timer will have to merely add a new
+`Task`, which the main thread will execute, to call the callback:
```

Comment:
```suggestion
`Task`, which the primary thread will execute, to call the callback:
```

Reply:
(can't say "main" or it would be inconsistent with later in the chapter)

Reply:
Great catch!